[Monk] Mistweaver implementation

acornellier opened the pull request.

Commit:
mistweaver functional

rebase:
TODO: remove shared totm

Commit:
awakened jadefire tiger palm proc

Commit:
ancient concordance blackout kick proc

Commit:
simplify jadefire tiger palm

Commit:
ancient concordance buff

Commit:
clang-format

Commit:
add a comment to ancient concordance

renanthera commented on engine/class_modules/monk/sc_monk.cpp:
```diff
@@ -1186,6 +1186,7 @@ struct tiger_palm_t : public overwhelming_force_t<monk_melee_attack_t>
 {
   bool face_palm;
   action_t *tigers_ferocity;
+  action_t *tiger_palm_jadefire_proc;
```
This is no longer used.

renanthera commented on engine/class_modules/monk/sc_monk.cpp:
```diff
@@ -1767,6 +1771,16 @@ struct blackout_kick_t : overwhelming_force_t<charred_passions_t<monk_melee_atta
+    // The Ancient Concordance buff's spell data modified number of targets hit, but the value is 1 instead of 3, so it
+    // is useless. We set the value manually to 3.
+    if ( p()->buff.ancient_concordance->check() )
+      return 3;
```
It's not necessarily useless, as it may provide 2 stacks in game. Note that the effect has the "Aura Points on Client" bit set, which has to do with behaviours related to effect stacking.

Would need to perform some in game science, as it is also not logged.

renanthera commented on engine/class_modules/monk/sc_monk.cpp:
```diff
@@ -7313,8 +7326,12 @@ void monk_t::init_spells()
+  else if ( baseline.mistweaver.teachings_of_the_monastery->ok() )
+    shared.teachings_of_the_monastery = baseline.mistweaver.teachings_of_the_monastery;
+  else
+    shared.teachings_of_the_monastery = spell_data_t::not_found();
```
If you're not comfortable writing a new equivalent function to `_priority` that is more friendly to multiple parameter types, let me know and I can put something together.

This feels gross, even I'm generally trying to get away from shared spell pointers.

acornellier replied:
sure, no idea how to rewrite `_priority` tbh

Commit:
remove unused member variable

Commit:
secret infusion talent + 4 buffs

Commit:
.percent()

Commit:
dance of chi-ji

Commit:
disable some effects on Heart of the Jade Serpent

HotJS effects 3, 4, and 7 duplicate the reduced CD from effect 2 for Renewing Mist, Life Cocoon, and TFT.

Commit:
fix enveloping mist

Commit:
improve apl

Commit:
apparently jfs does more damage than sck

Commit:
Revert "disable some effects on Heart of the Jade Serpent"

This reverts commit 3c2b671308540cdb739a7c401f8a952dbdf523e4.

Commit:
buff_t::set_default_value check if data is ok before setting values

no longer changes global fallback buff and stops assertion from failing

Commit:
update mw todo

Commit:
sheilun's gift

Commit:
fix hotjs max stacks with veil of pride

Commit:
no jadefire stomp past 10 targets

Commit:
jadefire stomp reset

Commit:
fix stomp reset condition

Commit:
remove zen pulse

Commit:
kate cr

Commit:
remove mistweaver-only healer stuff, and old commented hotfixes

Commit:
more cleanup, save the manatee

renanthera committed:
[CI] Add Mistweaver to CI, allow experimental specializations to be run.

renanthera committed:
[monk] Restore vivify `parse_options`. Remove `shared.shadowboxing_treads`.

Commit:
fix set_default_value assert

Commit:
vivify consumes august dynasty

renanthera committed:
[monk] Swap August Dynasty to `CONSUME_BUFF` instead of manual expiration.

renanthera commented:
The only things left to double check are TP OF working (I'm unsure if that's resolved) and whether or not all of the appropriate actions are using SP conversion properly when MW.

At that point, I would consider this good to merge (even prior to SEF completion).

A more flexible `_priority` is certainly a TODO, but not mandatory. It's in the same category of priority as my intended spell data storage changes.

Commit:
bring pack totm buff trigger

Commit:
xuens guidance can proc independently for each stack of totm

renanthera committed:
[monk] Fix OF for Tiger Palm.

Commit:
remove ancient concordance

renanthera committed:
[monk] Fix errors in rebase.

renanthera committed:
empty
